### ISSUE TITLE
Add a Lorentz training and evaluation example

### ISSUE THREAD

**Issue by t-yamsaki** [t-yamsaki/neembed#48]
## Goal

Add a small runnable example demonstrating that the existing neembed workflow works with `manifold="lorentz"` without introducing Lorentz-specific trainer or evaluator APIs.

## Scope

Add one concise example using the same public classes introduced before v0.3:

```python
from neembed import (
    ManifoldEmbeddingEvaluator,
    ManifoldSentenceTransformer,
    ManifoldMultipleNegativesRankingLoss,
    ManifoldTrainer,
)
```

The example should show:
- Lorentz model construction
- a tiny hierarchy / aligned-pair dataset
- ordinary training with `ManifoldTrainer`
- epoch or post-training evaluation with `ManifoldEmbeddingEvaluator`
- geodesic-distance inference
- the Lorentz ambient output shape relative to `embedding_dim`

Reuse the existing batch contract and avoid duplicate positives within a multiple-negatives batch.

## Acceptance criteria

- [ ] example is runnable from the repository with one documented command
- [ ] example uses `manifold="lorentz"`
- [ ] no geometry-specific trainer, loss, evaluator, Dataset, or DataLoader abstraction is introduced
- [ ] training loss and evaluation metrics are finite
- [ ] the example makes intrinsic vs ambient dimension behavior discoverable
- [ ] a lightweight smoke test verifies the example contract without downloading a model where practical
- [ ] the example does not claim Lorentz is generally better than Poincaré

## Out of scope

- large datasets
- hyperparameter tuning
- notebooks
- new metrics
- new losses
- benchmark claims

## Dependency

Implement after #46. #47 should also be complete or sufficiently stable so the example is built on validated geometry semantics.
